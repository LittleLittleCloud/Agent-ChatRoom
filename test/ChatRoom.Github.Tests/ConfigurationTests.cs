using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using ApprovalTests;
using ChatRoom.OpenAI;
using FluentAssertions;
using Json.Schema;
using Xunit;
using Json.Schema.Generation;

namespace ChatRoom.Github.Tests;

public class ConfigurationTests
{
    [Fact]
    [UseReporter(typeof(DiffReporter))]
    [UseApprovalSubdirectory("ApprovalTests")]
    public void VerifyConfigurationSchema()
    {
        var schema = new JsonSchemaBuilder()
            .FromType<GithubConfiguration>()
            .Build();
        var json = JsonSerializer.Serialize(schema, new JsonSerializerOptions { WriteIndented = true });
        Approvals.Verify(json);

        var schemaFileName = "chatroom_github_configuration_schema.json";
        var schemaFilePath = Path.Join("Schema", schemaFileName);
        var schemaFile = File.ReadAllText(schemaFilePath);
        schemaFile.Should().BeEquivalentTo(json);
    }
}
