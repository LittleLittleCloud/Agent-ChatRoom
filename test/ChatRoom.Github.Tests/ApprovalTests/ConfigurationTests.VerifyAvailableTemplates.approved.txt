templates

templates[0] = {
    "$schema": "chatroom_github_configuration_schema.json",
    "issue_helper": {
        "openai_config": {
            "llm_type": "OpenAI",
            "openai_model_id": "gpt-3.5-turbo",
            "openai_api_key": "<your-api-key>"
        }
    },
    "github_repo_name": "<repo-name>",
    "github_repo_owner": "<repo-owner>",
    "github_token": "<gh-token>"
}

